### ISSUE TITLE
Should quickstarts-base-url links in the current documentation go to matching release branch instead of main

### ISSUE THREAD

**Issue by rolfedh** [quarkusio/quarkus#41817]
### Description

In the current documentation, all quickstarts-base-url attribute values are configured to link to the main branch. However, the quickstarts in the main branch may contain features that are not supported in the release the user is using. To ensure compatibility and avoid potential issues, should we consider redefining these attributes to link to the corresponding release branch?

### Implementation ideas

One approach might be to define the release branch value in the attributes file, like this:
```
quickstarts-clone-url: -b {github-branch} ${quickstarts-base-url}.git
:quickstarts-archive-url:  ${quickstarts-base-url}/archive/{github-branch}
```
Another approach might be to remove branch definitions from the attributes and place them only in the content files, like this:
```
Make sure you have the getting-started application at hand, or clone the Git repository: `git clone -b {github-branch} {quickstarts-clone-url}`, or download an {quickstarts-archive-url}{github-branch}[archive]. The solution is located in the `getting-started` directory.
```

**Comment by gsmet** [comment]
> In the current documentation, all quickstarts-base-url attribute values are configured to link to the main branch.

Are they?

They point to the main branch for the current active branch. They point to a specific branch for LTS releases, typically in https://quarkus.io/version/3.8/guides/validation .

I don't think there's anything to fix on the community side for this. Or please be more specific.

**Comment by michalvavrik** [comment]
Yes, I think there has been confusion because it's correct in a community and what is explained here was only true for a product.

**Comment by gsmet** [comment]
OK, so let's close this then.

**Comment by rolfedh** [comment]
> They point to the main branch for the current active branch. They point to a specific branch for LTS releases, typically in https://quarkus.io/version/3.8/guides/validation .

I see this now. Thanks for considering.

**Comment by rolfedh** [comment]
One of the QE reviewers requested a change in the community `/asciidoc` files that had been synchronized to the product repository. I was following through to ensure this request didn't get overlooked. Upon inspecting the community attributes files, I didn't see an affordance for the version number. When I reviewed the community `3.12` docs, I noticed that the links pointed to `main` instead of `3.12`. However, had I inspected the `3.8` docs, I would have seen that the links correctly pointed to `3.8`. Thank you for the conversation.
